Assets/Scripts/Algorithms/Patrolling/Components/PartitionDistributionComponent.cs

```diff
@@ -0,0 +1,85 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+using Maes.Map;
+using Maes.Robot;
+
+using UnityEngine;
+
+namespace Maes.Algorithms.Patrolling.Components
+{
+    public sealed class PartitionDistributionComponent : IComponent
+    {
+        private readonly IRobotController _robotController;
+
+        public int PreUpdateOrder => -10000;
+
+        public int PostUpdateOrder => -10000;
+
+        public static List<Partition> Partitions { get; private set; } = new List<Partition>();
```

Comment:
Why do we have this?

Reply:
The component initializes and keeps track of the partitions.
Theyre used for redistribution, we could probably get away with it being one component, but i think there's a good chance we will use partitions in other components as well.